.buildkite/scripts/run_scala_task.sh

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env bash
+<<EOF
+Run a Scala task in Buildkite.  This includes:
+
+  - running tests
+  - building a Docker image (if applicable)
+  - publishing the Docker image to ECR (if on main)
+
+The script will try to be intelligent about what to do, e.g. deciding whether
+to publish images or whether tests need docker-compose.
+
+This script is mirrored in the catalogue-pipeline and storage-service repos.
+
+== Usage examples ==
+
+    run_scala_task.sh bag_verifier
+
+        This will run the tests for the 'bag_verifier' app, which is
+        in the 'bag_verifier' folder in the root of the repo.
+        It will build the image on pull requests, and on pushes to main
+        it will also publish the image to ECR.
+
+    run_scala_task.sh snapshot_generator snapshots/snapshot_generator
+
+        This runs the tests for the 'snapshot_generator' app, which is
+        in the 'snapshots/snapshot_generator' folder.  It also builds and
+        publishes Docker images on pull requests/pushes to main, respectively.
+
+        This second argument is useful in larger repositories where not
+        every app is in the top level.
+
+    run_scala_task.sh display --is-library
+
+        This runs the tests for the 'display' library, which is in the
+        'display' folder in the root of the repo.  It does not build or
+        publish Docker images, because we don't have them for libraries.
+
+    run_scala_task.sh pipeline_storage common/pipeline_storage --is-library
+
+        This runs the tests for the 'pipeline_storage' library, which is
+        in the 'common/pipeline_storage' folder in the root of the repo.
+        It doesn't build or publish Docker images.
+
+        This second argument is useful in larger repositories where not
+        every library is in the top level.
+
+EOF
+
+set -o errexit
+set -o nounset
+
+ECR_REGISTRY="760097843905.dkr.ecr.eu-west-1.amazonaws.com"
+
+parse_args() {
+  if (( $# == 1 ))
+  then
+    PROJECT_NAME="$1"
+    PROJECT_DIRECTORY="$1"
+    PROJECT_TYPE="app"
+  elif (( $# == 3 )) && [[ "$3" == "--is-library" ]]
+  then
+    PROJECT_NAME="$1"
+    PROJECT_DIRECTORY="$2"
+    PROJECT_TYPE="library"
+  elif (( $# == 2 )) && [[ "$2" == "--is-library" ]]
+  then
+    PROJECT_NAME="$1"
+    PROJECT_DIRECTORY="$1"
+    PROJECT_TYPE="library"
+  elif (( $# == 2))
+  then
+    PROJECT_NAME="$1"
+    PROJECT_DIRECTORY="$2"
+    PROJECT_TYPE="app"
+  else
+    echo "Usage: publish_sbt_app.sh <PROJECT> [<PROJECT_DIRECTORY> | --is-library]" >&2
+    exit 1
+  fi
```

Comment:
😱 aren't bash scripts wonderful - is there a better way of doing this?

Reply:
Yes, I’m working on that now.

Reply:
I found a way to get sbt to dump the project directory, which simplifies this two ways:

1. We don't need to supply a PROJECT_DIRECTORY argument, because sbt will tell us
2. We don't need to check if this is an app/library, we just look for a Dockerfile in that directory

So now it has a much simpler interface.